pkg/apis/core/types_shoot.go

```diff
@@ -233,8 +233,8 @@ type ShootObservabilityRotation struct {
 	LastCompletionTime *metav1.Time
 }
 
-// ShootServiceAccountKeyRotation contains information about the service account key credential rotation.
-type ShootServiceAccountKeyRotation struct {
+// ServiceAccountKeyRotation contains information about the service account key credential rotation.
+type ServiceAccountKeyRotation struct {
```

Comment:
I didn't move this here because only `core/v1beta1/types_shoot.go` would be an appropriate fit (since this only in this file "common" types are). The types in `core/types_shoot.go` and `core/v1alpha1/types_shoot.go` are not reused anywhere.

Reply:
This is being resued for the garden object also.

Reply:
Sorry? The types in `core/types_shoot.go` and `core/v1alpha1/types_shoot.go` are not reused anywhere, are they?

Reply:
In this file https://github.com/gardener/gardener/blob/749cb7b70b3e83a7bf18984b82e898de48a31744/pkg/apis/operator/v1alpha1/types.go `ServiceAccountKey` field is of type `ServiceAccountKeyRotation`. `ServiceAccountKeyRotation` is defined in `type_shoot.go`.

Reply:
Yes. Please read my argument in https://github.com/gardener/gardener/pull/7573#discussion_r1124115180 again. The `Garden` API only reuses the fields from `core/v1beta1/types_shoot.go`, but those in `core/v1alpha1/types_shoot` and `core/types_shoot.go` are not reused. Hence, moving all this to `types_common.go` is somewhat inconsistent.